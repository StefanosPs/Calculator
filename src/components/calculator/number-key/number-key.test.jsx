import React from 'react';
import renderer from 'react-test-renderer';

import {
    numbersKeysArray
  } from "../../../classes/calculator/calculatorClass";

import NumberKey from './number-key';

it('renders NumberKey when there are no props', () => {
  const tree = renderer.create(<NumberKey />).toJSON();
  expect(tree).toMatchSnapshot(); 
});

it('renders NumberKey when there are props', () => {
    const numberKeyFull = renderer.create(<NumberKey numbersKeysArray={numbersKeysArray} />).toJSON();
    expect(numberKeyFull).toMatchSnapshot();
});