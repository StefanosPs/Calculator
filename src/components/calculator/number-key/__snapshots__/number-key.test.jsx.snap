// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NumberKey when there are no props 1`] = `
<div
  className="container"
>
  <div
    className="justify-content-center row"
  >
    <div
      className="col"
    >
      <div
        className="fade alert alert-danger show"
        role="alert"
      >
        Prop numbersKeysArray must be an array
      </div>
    </div>
  </div>
</div>
`;

exports[`renders NumberKey when there are props 1`] = `
<div
  className="container"
>
  <div
    className="justify-content-center row"
  >
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow mb-1 pl-3 pr-3 text-white  rounded btn btn-outline-dark"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </div>
</div>
`;
